wagtail/core/models.py

```diff
@@ -1830,8 +1976,45 @@ def is_latest_revision(self):
         latest_revision = PageRevision.objects.filter(page_id=self.page_id).order_by('-created_at', '-id').first()
         return (latest_revision == self)
 
-    def publish(self):
+    def publish(self, user=None, changed=True, log_action=True, previous_revision=None):
+        """
+        Publishes or schedules revision for publishing.
+
+        :param user: the publishing user
+        :param changed: indicated whether content has changed
+        :param log_action:
+            flag for the logging action. Pass False to skip logging. Cannot pass an action string as the method
+            performs several actions: "publish", "revert" (and publish the reverted revision),
+            "schedule publishing with a live revision", "schedule revision reversal publishing, with a live revision",
+            "schedule publishing", "schedule revision reversal publishing"
+        :param previous_revision: indicates a revision reversal. Should be set to the previous revision instance
+        """
         page = self.as_page_object()
+
+        def log_scheduling_action(revision, previous_revision=None, user=None, changed=changed):
+            if not previous_revision:
+                action = 'wagtail.schedule.publish'
+                revision_for_data = self
+            else:
+                action = 'wagtail.schedule.revert'
```

Comment:
Do we need a separate action for scheduling a revert? The revert has already been logged as `wagtail.revert`

Reply:
we do for publish, so it makes sense for this

Reply:
Oh, I didn't realise we did it for publishing too. What's the advantage of having a separate revert+publish event? We don't have a separate event for edit+publish do we?

Reply:
dropped `wagtail.schedule.revert` in favour of a single `wagtail.publish.schedule`